ext/ddtrace.c

```diff
@@ -1367,6 +1367,131 @@ static void dd_disable_if_incompatible_sapi_detected(void) {
     }
 }
 
+static php_stream_wrapper_ops traced_http_wrapper_wops;
+static php_stream_wrapper_ops traced_https_wrapper_wops;
+static php_stream_wrapper ddtrace_original_http_wrapper;
+static php_stream_wrapper ddtrace_original_https_wrapper;
+static php_stream_wrapper ddtrace_traced_http_wrapper;
+static php_stream_wrapper ddtrace_traced_https_wrapper;
+
+static php_stream *ddtrace_stream_opener(
+    php_stream_wrapper *wrapper,
+    const char *filename,
+    const char *mode,
+    int options,
+    zend_string **opened_path,
+    php_stream_context *context,
+    int flags,
+    const char *persistent_id,
+    uint32_t persistent_id_len,
+    const char *wrapper_data,
+    uint32_t wrapper_data_len
+) {
+    if (!context) {
+        context = php_stream_context_alloc();
+    }
+
+    // Inject distributed tracing headers
+    zval *http_context_zv = zend_hash_str_find(Z_ARRVAL(context->options), "http", sizeof("http") - 1);
+    if (!http_context_zv) {
+        zval tmp;
+        array_init(&tmp);
+        zend_hash_str_update(Z_ARRVAL(context->options), "http", sizeof("http") - 1, &tmp);
+        http_context_zv = zend_hash_str_find(Z_ARRVAL(context->options), "http", sizeof("http") - 1);
```

Comment:
```suggestion
        http_context_zv = zend_hash_str_find_deref(Z_ARRVAL(context->options), "http", sizeof("http") - 1);
```
Might be reference, we need to guard against that. Also in the find above.